ybdata/commands/convert.py

```diff
@@ -80,9 +80,9 @@ def handle(self, args):
         # Perform the conversion and save
         if dtype == ".npz":
             # Convert to the appropriate array type
-            if args.X_dtype is not None:
+            if hasattr(args, 'X_dtype'):
                 X = X.astype(args.X_dtype)
-            if args.y_dtype is not None:
+            if hasattr(args, 'y_dtype'):
```

Comment:
So these original conditionals were intended to allow for datatype conversions and unfortunately, they were not updated in the args list (which appears to remain just `dtype`). The idea was to convert the entire array as the specified datatype, one of [`"biufcmMOSUV"`](https://www.numpy.org/devdocs/reference/generated/numpy.dtype.kind.html) - a character code for different types. Although this doesn't work for heterogeneous datasets, most YB datasets are single types.  As you may have need of these arguments to encode your dataset, would you mind making the fix? 

Basically on line 40 this:

```python
args = {
        ("-d", "--dtype"): {
            "default": None, "metavar": "T", "type": str,
            "help": "convert to type (one of `biufcmMOSUV`) when dst is npz",
        },
        "src": {
            "type": str, "metavar": "src",
            "help": "path to dataset source file to convert",
        },
        "dst": {
            "type": str, "metavar": "dst",
            "help": "path to write converted dataset to",
        },
    }
```

Needs to be changed to:

```python
args = {
        ("-X", "--X-dtype"): {
            "default": None, "metavar": "T", "type": str,
            "help": "convert X to type (one of `biufcmMOSUV`) when dst is npz",
        },
        ("-y", "--y-dtype"): {
            "default": None, "metavar": "T", "type": str,
            "help": "convert y to type (one of `biufcmMOSUV`) when dst is npz",
        },
        "src": {
            "type": str, "metavar": "src",
            "help": "path to dataset source file to convert",
        },
        "dst": {
            "type": str, "metavar": "dst",
            "help": "path to write converted dataset to",
        },
    }
```

You can then revert the conditionals here to their original specification.

Reply:
Made this change in the latest commit.

Reply:
Thank you!